src/nvim/tui/input.c

```diff
@@ -124,16 +125,34 @@ static void tinput_wait_enqueue(void **argv)
     const String keys = { .data = buf, .size = len };
     if (input->paste) {
       String copy = copy_string(keys);
-      multiqueue_put(main_loop.events, tinput_paste_event, 3,
-                     copy.data, copy.size, (intptr_t)input->paste);
+      if (ui_client_channel_id) {
+        Array args = ARRAY_DICT_INIT;
+        ADD(args, STRING_OBJ(copy_string(keys)));  // 'data'
```

Comment:
Will this leak memory?

Reply:
`rpc_send_event` consumes the array and frees it when done with it.

Reply:
I meant  `copy` is unused when this path is chosen.